File: src/me/krickl/memebotj/InternalCommands/ModeratorCommands/ChannelInfoCommand.scala
package me.krickl.memebotj.InternalCommands.ModeratorCommands

import java.io.IOException
import java.nio.charset.Charset
import java.nio.file.{Paths, Files}

import me.krickl.memebotj._
import me.krickl.memebotj.Utility.CommandPower

class ChannelInfoCommand(channel: String, command: String, dbprefix: String)
		extends CommandHandler(channel, command, dbprefix) {

	this.setNeededCommandPower(CommandPower.adminAbsolute)

	override def commandScript(sender: UserHandler, channelHandler: ChannelHandler, data: Array[String]) {
		try {
      if(data(0) == "reload") {
        Memebot.urlBanList = Files.readAllLines(Paths.get(Memebot.memebotDir + "/urlblacklist.cfg"),
          Charset.defaultCharset()).asInstanceOf[java.util.ArrayList[String]]

        Memebot.phraseBanList = Files.readAllLines(Paths.get(Memebot.memebotDir + "/phrasebanlist.cfg"),
          Charset.defaultCharset()).asInstanceOf[java.util.ArrayList[String]]

        Memebot.globalBanList = Files.readAllLines(Paths.get(Memebot.memebotDir + "/globalbanlist.cfg"),
          Charset.defaultCharset()).asInstanceOf[java.util.ArrayList[String]]

        channelHandler.sendMessage(f"Reloaded global ban lists. There are ${Memebot.urlBanList.size()} banned urls, ${Memebot.phraseBanList.size()} banned phrases and ${Memebot.globalBanList.size()} blacklisted users.")

      }
    } catch {
      case e: ArrayIndexOutOfBoundsException =>
        channelHandler.sendMessage(f"Is live: ${channelHandler.isLive} || Points per update ${channelHandler.getPointsPerUpdate} || Game: ${channelHandler.currentGame} || ${channelHandler.followAnnouncement}", this.getChannelOrigin)
      case e: IOException =>
        e.printStackTrace()
        channelHandler.sendMessage(e.toString)
    }
	}
}

File: src/me/krickl/memebotj/Commands/Internal/ChannelInfoCommand.java
package me.krickl.memebotj.Commands.Internal;

import me.krickl.memebotj.ChannelHandler;
import me.krickl.memebotj.Commands.CommandHandler;
import me.krickl.memebotj.Memebot;
import me.krickl.memebotj.UserHandler;
import me.krickl.memebotj.Utility.CommandPower;

import java.io.IOException;

/**
 * This file is part of memebotj.
 * Created by unlink on 11/04/16.
 */
public class ChannelInfoCommand extends CommandHandler {
    public ChannelInfoCommand(ChannelHandler channelHandler, String commandName, String dbprefix) {
        super(channelHandler, commandName, dbprefix);
    }

    @Override
    public void overrideDB() {
        this.setNeededCommandPower(CommandPower.adminAbsolute);
    }

    @Override
    public void commandScript(UserHandler sender, String[] data) {
        try {
            if (data[0].equals("reload")) {
                Memebot.readConfig();
            }
        } catch(ArrayIndexOutOfBoundsException e) {
            getChannelHandler().sendMessage("Is live: " + getChannelHandler().isLive() + " || Points per update: " + getChannelHandler().getPointsPerUpdate() + " || Game: " + getChannelHandler().getCurrentGame(), this.getChannelHandler().getChannel());
        }
    }
}
